datadog-oci-orm/metrics-setup/serviceconnector.tf

```diff
@@ -1,27 +1,46 @@
+locals {
+  # Variables to manage connector hubs
+  metrics_compartments    = toset(split(",", var.metrics_compartments))
+  supported_namespaces    = var.metrics_namespaces
+  compartment_tuple       = [for comp in local.metrics_compartments : { compartment = comp, namespaces = tolist(setintersection(local.supported_namespaces, [for ns in data.oci_monitoring_metrics.existing_namespaces[comp].metrics : ns.namespace])) }]
+  # Order by compartment id
```

Comment:
Given temp_derived_namespaces is a map, I don't think it will be ordered by compartment ID as this comment states.

Also I don't think temp_derived_namespaces is needed. I believe derived_namespaces can be created straight from the tuple:
```
  derived_namespaces      = [
    for tup in local.compartment_tuple : {
      compartment = tup.compartment,
      namespaces = tup.namespaces
    }
    if length(tup.namespaces) > 0
  ]
```